### PR TITLE
Stubs for slumber

### PR DESCRIPTION
**Pull request by exemplary-citizen**
A fairly popular object-oriented interface to ReSTful APIs. It acts as a wrapper around the requests library and abstracts away the handling of URLs, serialization, and request processing.

I started with stubgen, then cleaned up to address missing functions. Testing plan is to keep an eye on CI and fix any issues that arise.

cc: @JukkaL

### CODE REVIEW

**Comment by srittau** [comment]
This needs to be added to `pyrightconfig.stricter.json`.

File: stubs/slumber/slumber/exceptions.pyi [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from .utils import iterator as iterator, url_join as url_join
```
Comment (srittau) [inline stubs/slumber/slumber/exceptions.pyi:1]: stubgen is overzealous when creating re-exports. As these imports are not used in the stubs, they shouldn't be here. Re-exporting them is more than likely just an artifact of the implementation. Same for `serialize.pyi`.

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks, looks good. One comment about imports below.

File: stubs/slumber/slumber/exceptions.pyi (hunk 1 above)
Comment (exemplary-citizen) [reply stubs/slumber/slumber/exceptions.pyi]: Thanks for sharing that! makes a lot of sense.

**Review by srittau — APPROVED** [review]

**Pull request merged**